src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -208,6 +208,11 @@ public static String formatDate(Date date) {
     public static String formatTime12H(Date date) {
         if (date == null)
             return "";
+        Calendar c = Calendar.getInstance();
+        c.setTime(date);
+        if (c.get(Calendar.HOUR_OF_DAY) == 12 && c.get(Calendar.MINUTE) == 0) {
```

Comment:
I think we should show 12.01pm (or any x minutes after 12) as 12.01 NOON too? I think it's not sufficient just to handle the 12pm case. Otherwise, users may be confused about whether 12.01 pm is 1 minute after noon or midnight.